sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/parameters.scala

```diff
@@ -96,7 +96,11 @@ case class PosParameterizedQuery(child: LogicalPlan, args: Array[Expression])
  */
 object BindParameters extends Rule[LogicalPlan] with QueryErrorsBase {
   private def checkArgs(args: Iterable[(String, Expression)]): Unit = {
-    args.find(!_._2.isInstanceOf[Literal]).foreach { case (name, expr) =>
+    def isNotAllowed(expr: Expression): Boolean = expr.exists {
+      case _: Literal | _: CreateArray | _: MapFromArrays | _: CreateMap => false
```

Comment:
We should recursively check the children of `CreateArray`, etc.

Reply:
`_.exists` checks the children recursively, doesn't it? So, we allow mix of arrays, maps and literals.

Reply:
Let me add an negative test with an deep nested prohibited expression.

Reply:
Ah I see! It's `TreeNode#exists`

Reply:
Can we add `CreateStruct`? It's weird that we only leave it out.

Reply:
ok. How about to support `MapFromEntries` as well? And maybe `CreateNamedStruct`?

Reply:
`CreateNamedStruct` is the actual expression for `CreateStruct`. `MapFromEntries` also look fine to me.